### ISSUE TITLE
[bug]: Unable to start litd because of an error associated with a force close of a taproot asset channel

### ISSUE THREAD

**Issue by Liongrass** [lightninglabs/taproot-assets#1138]
### Background

I'm running `litd 0.13.994`. I have multiple channels, including taproot asset channels. Recently I have been unable to start litd due to an error associated with a taproot asset channel.

```
2024-09-23 19:33:49.767 [DBG] LNWL: ChannelPoint(1e68171eae852f3b6249084e2a186902baf227c9b3a0faf6505ea249a489eaaa:0): Restoring 0 dangling remote updates
2024-09-23 19:33:49.767 [DBG] LNWL: ChannelPoint(1e68171eae852f3b6249084e2a186902baf227c9b3a0faf6505ea249a489eaaa:0): Restoring 0 local updates that the peer should sign
2024-09-23 19:33:49.770 [INF] LNWL: Generating new musig2 sig for session=33c4849df75806da766db9133bfef4d6105dbd36558aa68fe60c4bd9c0fb9adc, nonces=NoncePair(verification_nonce=036168bff0c15de80d6ccfe8b169d0340c209e9c4acdb55f970e3f719f62b6d1580208f056a58924d8d602569e2a7ca898386db7e430904883abe3b2208a525abf5f, signing_nonce=036e99122b0c453d968d764b08a9e0ffcdee6942dc05dc8d5e6c8dca4d47c01f6b027cb5f6521b6728ee4d8009706986cbe9c25ae0d143d69cab38e25f6926b3a476)
2024-09-23 19:33:49.773 [ERR] CNCT: ChannelArbitrator(1e68171eae852f3b6249084e2a186902baf227c9b3a0faf6505ea249a489eaaa:0): unable to force close: unable to combine musig partial sigs: error combining partial signature: error combining partial signature: final signature is invalid
2024-09-23 19:33:49.773 [ERR] CNCT: ChannelArbitrator(1e68171eae852f3b6249084e2a186902baf227c9b3a0faf6505ea249a489eaaa:0): unable to advance state: unable to combine musig partial sigs: error combining partial signature: error combining partial signature: final signature is invalid
2024-09-23 19:33:49.776 [INF] CNCT: ChainArbitrator shutting down...
```

I've attached the complete logs below.
[2024-09-23_litd 0.13.994.log](https://github.com/user-attachments/files/17153982/2024-09-23_litd.0.13.994.log)

**Comment by MegalithicBTC** [comment]
I'm pretty sure I have this issue here. You can grab the full logs here: https://github.com/lightningnetwork/lnd/discussions/9152

**Comment by dstadulis** [comment]
@jharveyb received new debugging information from @Liongrass
Let's update current root-cause theories for these symptoms

**Comment by dstadulis** [comment]
> I'm pretty sure I have this issue also. You can grab the full logs here: [lightningnetwork/lnd#9152](https://github.com/lightningnetwork/lnd/discussions/9152) ... it's a testnet node and I'm happy to share the database file if that's useful

@MegalithicBTC could you share the DB?

**Comment by gijswijs** [comment]
I took a closer look at @Liongrass 's logs.
During startup litd tries to craft a backup for 10 channels:
```
2024-09-23 21:09:38.746 [DBG] CHBU: Crafting backup for ChannelPoint(77c85850046d033adc333659fe45b8837392bf54815c96c4cd114178e0402ab5:0)
2024-09-23 21:09:38.747 [DBG] CHBU: Crafting backup for ChannelPoint(d29379c9982b6d262f5769458da2aa846a6ed0c8a72f8e68e7a439c4592d3f1e:0)
2024-09-23 21:09:38.747 [DBG] CHBU: Crafting backup for ChannelPoint(f19735394f3fcdc887b037c3d2d4b4a8ef594ff45fb31b36b1d2d0e03ccea2a3:0)
2024-09-23 21:09:38.747 [DBG] CHBU: Crafting backup for ChannelPoint(6d9d4335c0509e717e6b5701550cf79e5ede29d15ab94f73b9286a85dfefdb3e:0)
2024-09-23 21:09:38.747 [DBG] CHBU: Crafting backup for ChannelPoint(3fa93d25f99605fe9f76c4492f6ea3e35c27a75b57fe79792862c4b713dcd609:0)
2024-09-23 21:09:38.747 [DBG] CHBU: Crafting backup for ChannelPoint(5737f8bf578ac0014501ec8ed4ecf3ebdf1a6dc347850158cad033a6a69d5825:1)
2024-09-23 21:09:38.747 [DBG] CHBU: Crafting backup for ChannelPoint(1450a888f4990264700bb372a9e261816128df0ab4b5b84b30e387915a67e066:0)
2024-09-23 21:09:38.747 [DBG] CHBU: Crafting backup for ChannelPoint(b9740e90782497f9119fbbf4a787b1bf044e480813d781f402679a1ede96043c:0)
2024-09-23 21:09:38.748 [DBG] CHBU: Crafting backup for ChannelPoint(1e68171eae852f3b6249084e2a186902baf227c9b3a0faf6505ea249a489eaaa:0)
2024-09-23 21:09:38.748 [DBG] CHBU: Crafting backup for ChannelPoint(99b47ab0635779123cc106e12c94de64691e0e574bf32d9e7a5c0b6d3e24cfc0:0)
```

For 4 of them it can't seem to find the confirmation details:
```
2024-09-23 21:10:41.362 [DBG] NTFN: Failed getting conf details from index (unable to query for txid d29379c9982b6d262f5769458da2aa846a6ed0c8a72f8e68e7a439c4592d3f1e: -5: No such mempool transaction. Use -txindex or provide a block hash to enable blockchain transaction queries. Use gettransaction for wallet transactions.), scanning manually
2024-09-23 21:10:41.363 [DBG] NTFN: Failed getting conf details from index (unable to query for txid 1e68171eae852f3b6249084e2a186902baf227c9b3a0faf6505ea249a489eaaa: -5: No such mempool transaction. Use -txindex or provide a block hash to enable blockchain transaction queries. Use gettransaction for wallet transactions.), scanning manually
2024-09-23 21:10:41.445 [DBG] NTFN: Failed getting conf details from index (unable to query for txid 99b47ab0635779123cc106e12c94de64691e0e574bf32d9e7a5c0b6d3e24cfc0: -5: No such mempool transaction. Use -txindex or provide a block hash to enable blockchain transaction queries. Use gettransaction for wallet transactions.), scanning manually
2024-09-23 21:10:41.447 [DBG] NTFN: Failed getting conf details from index (unable to query for txid f19735394f3fcdc887b037c3d2d4b4a8ef594ff45fb31b36b1d2d0e03ccea2a3: -5: No such mempool transaction. Use -txindex or provide a block hash to enable blockchain transaction queries. Use gettransaction for wallet transactions.), scanning manually
```

The daemon then decides to force close `1e68171eae852f3b6249084e2a186902baf227c9b3a0faf6505ea249a489eaaa`, but I think either one of those 4 channels would have resulted in the same error. In any case the force close trips up during musig2.

Will investigate further.

**Comment by gijswijs** [comment]
@Liongrass Did you have 10 channels active? Or maybe just 6?

**Comment by guggero** [comment]
@Liongrass do you remember with what version of `litd` you opened that channel? Was it with `v0.13.994-experimental` or earlier?

Anyway, I'm pretty sure this is probably what was fixed with https://github.com/lightningnetwork/lnd/pull/9046 which was only added to `litd` with `v0.13.995-experimental`.

So it's probably easiest to just abandon that channel (if you can run `abandonchannel` before it crashes). Or just abandon the node after all.

**Comment by guggero** [comment]
Discussed this internally and we're pretty confident this should've been fixed in `v0.13.995-experimental`.
So since I assume this is testnet, our suggestion is to just remove the `channel.db` and create new channels.

Closing the issue unless someone can confirm this still happens with channels opened on `v0.13.995-experimental` (or later).

**Comment by dstadulis** [comment]
>since I assume this is testnet, our suggestion is to just remove the channel.db and create new channels.

Confirmed in logs https://github.com/lightninglabs/taproot-assets/issues/1138 was on testnet
2024-09-23 21:09:26.535 [INF] LTND: Active chain: Bitcoin (network=testnet)
